vmdb/app/views/dashboard/_widget.html.haml

```diff
@@ -1,26 +1,26 @@
 - # Parameters:
 - # widget MiqWidget object
-%div{:id => "w_#{widget.id}"}
+%div{:id => "w_#{presenter.widget.id}"}
   #test.modbox
     %h2.modtitle{:style => "cursor:move"}
-      %a{:class => "#{widget.content_type}box"}
+      %a{:class => "#{presenter.widget.content_type}box"}
       - if role_allows(:feature => "dashboard_add")
-        = render(:partial => 'button_close', :locals => {:widget => widget})
-      = render(:partial => 'button_minmax', :locals => {:widget => widget})
-      - if ["report","chart"].include?(widget.content_type) &&
-      - !widget.contents_for_user(session[:userid]).blank?
-        = render(:partial => 'button_fullscreen', :locals => {:widget => widget})
-        = render(:partial => 'button_pdf', :locals => {:widget => widget})
-      - if ["chart"].include?(widget.content_type) &&
-      - !widget.contents_for_user(session[:userid]).blank?
-        = render(:partial => 'button_zoom', :locals => {:widget => widget})
+        = presenter.button_close
+      = presenter.button_minmax
+      - if %(report chart).include?(presenter.widget.content_type) &&
+      - presenter.widget.contents_for_user(session[:userid]).present?
+        = presenter.button_fullscreen
+        = presenter.button_pdf
+      - if %(chart).include?(presenter.widget.content_type) &&
+      - presenter.widget.contents_for_user(session[:userid]).present?
```

Comment:
maybe put this part of the condition one level up to make it clear that if there's no content for user, we don't draw anything